README.md

```diff
@@ -4,6 +4,37 @@
 
 This gem is an implementation of the [language server protocol specification](https://microsoft.github.io/language-server-protocol/) for Ruby, used to improve editor features.
 
+# Overview
+
+The intention of Ruby LSP is to provide a fast, accurate, and robust coding environment for Ruby developers, especially for large codebases.
+
+It's part of a [wider Shopify goal](https://github.com/Shopify/vscode-shopify-ruby) to provide a state-of-the-art experience to Ruby developers using modern standards for cross-editor features, documentation and debugging.
+
+It provides many features, including:
+
+* Syntax highlighting
+* Linting and formatting
+* Code folding
+* Selection ranges
+
+When used on typed Sorbet code, additional features are available:
+
+* Quick documentation lookup when hovering over a Ruby or Rails method.
+* Support for Go To Definition.
+
+Planned future features include:
+
+* Context-aware auto-completion and navigation ("Go To Definition") for non-typed code ([prototype](https://github.com/Shopify/ruby-lsp/pull/429))
+* Support for plug-ins to extend behavior
+
+ruby-lsp does not perform any type-checking or provide any type-related assistance, but it can be used alongside [Sorbet](https://github.com/sorbet/sorbet)'s LSP server.
+
+At the time of writing, these are the major differences between Ruby LSP and [Solargraph](https://solargraph.org/):
+
+* Solargraph [uses](https://solargraph.org/guides/yard) YARD documentation to gather information about your project and its gem dependencies. This provides functionality such as context-aware auto-completion and navigation ("Go To Definition")
+* Solargraph can be used as a globally installed gem, but Ruby LSP must be added to the Gemfile or gemspec. (There are pros and cons to each approach)
```

Comment:
Similar to my other comment, I'd point out that this is only true for projects using RuboCop.